### DISCUSSION TITLE
How to deploy the migrated project?

### DISCUSSION THREAD

**Discussion by mrgg2405** [CoreForms/docs#4] [Q&A]
Hi CoreForms Team,

I'm working on the migration of ASP.NET webform project to NETCore. As of now, I have successfully migrated the web forms with the help of CoreForms. When I was trying to publish the project and deploy it to local IIS, I encountered some issues:

1. I got 404 error because the Index page didn't exist in the publish folder.
2. I got a strange error as below after I copied the Index.aspx to the publish folder.
---
Server Error in '/' Application.

'vb' is not a supported language.
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.Web.HttpException: 'vb' is not a supported language.

Source Error:
An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace:
[HttpException (0x80004005): 'vb' is not a supported language.]
   System.Web.Configuration.CompilationSection.GetCompilerInfoFromLanguage(String language) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Configuration\CompilationSection.cs:642
   System.Web.Compilation.CompilationUtil.GetDefaultLanguageCompilerInfo(CompilationSection compConfig, VirtualPath configPath) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\CompilationUtil.cs:86
   System.Web.UI.TemplateParser.HandlePostParse() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\TemplateParser.cs:1244
   System.Web.UI.TemplateControlParser.HandlePostParse() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\TemplateControlParser.cs:298
   System.Web.UI.PageParser.HandlePostParse() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\PageParser.cs:212
   System.Web.UI.TemplateParser.Parse() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\TemplateParser.cs:689
   System.Web.UI.TemplateParser.Parse(ICollection referencedAssemblies, VirtualPath virtualPath) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\TemplateParser.cs:670
   System.Web.Compilation.BaseTemplateBuildProvider.get_CodeCompilerType() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BaseTemplateBuildProvider.cs:84
   System.Web.Compilation.BuildProvider.GetCompilerTypeFromBuildProvider(BuildProvider buildProvider) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildProvider.cs:412
   System.Web.Compilation.BuildProvidersCompiler.ProcessBuildProviders() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildProvidersCompiler.cs:116
   System.Web.Compilation.BuildProvidersCompiler.PerformBuild() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildProvidersCompiler.cs:208
   System.Web.Compilation.BuildManager.CompileWebFile(VirtualPath virtualPath) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildManager.cs:1710
   System.Web.Compilation.BuildManager.GetVPathBuildResultInternal(VirtualPath virtualPath, Boolean noBuild, Boolean allowCrossApp, Boolean allowBuildInPrecompile, Boolean throwIfNotFound, Boolean ensureIsUpToDate) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildManager.cs:1636
   System.Web.Compilation.BuildManager.GetVPathBuildResultWithNoAssert(HttpContext context, VirtualPath virtualPath, Boolean noBuild, Boolean allowCrossApp, Boolean allowBuildInPrecompile, Boolean throwIfNotFound, Boolean ensureIsUpToDate) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildManager.cs:1555
   System.Web.Compilation.BuildManager.GetVirtualPathObjectFactory(VirtualPath virtualPath, HttpContext context, Boolean allowCrossApp, Boolean throwIfNotFound) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildManager.cs:3192
   System.Web.Compilation.BuildManager.GetVirtualPathObjectFactory(VirtualPath virtualPath, HttpContext context, Boolean allowCrossApp) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildManager.cs:3160
   System.Web.Compilation.BuildManager.CreateInstanceFromVirtualPath(VirtualPath virtualPath, Type requiredBaseType, HttpContext context, Boolean allowCrossApp) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\Compilation\BuildManager.cs:3135
   System.Web.UI.PageHandlerFactory.GetHandlerHelper(HttpContext context, String requestType, VirtualPath virtualPath, String physicalPath) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\PageHandlerFactory.cs:62
   System.Web.UI.PageHandlerFactory.System.Web.IHttpHandlerFactory2.GetHandler(HttpContext context, String requestType, VirtualPath virtualPath, String physicalPath) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\UI\PageHandlerFactory.cs:54
   System.Web.HttpApplication.MapHttpHandler(HttpContext context, String requestType, VirtualPath path, String pathTranslated, Boolean useAppConfig) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\HttpApplication.cs:1308
   System.Web.MapHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\HttpApplication.cs:2829
   System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\HttpApplication.cs:2040
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) in C:\Build\a6d07cf9277dedfe\src\CoreForms.Web\HttpApplication.cs:2064

Version Information: .NET 7.0.19 / CoreForms 7.0.0-alpha+.Sha.86dc442252ca2db23998937356a7de8159530ac7

---

I haven't found any document regarding how to publish the migrated project, so I hope I can get some guidelines of how to do (especially for ASPX pages) from the team. Thank you very much.

**Comment by MichaelKetting** [comment]
Hi @mrgg2405 
You got a couple of questions in there, I'll try to catch them all:

1. VB: at this time, we don't support VB.Net when compiling ASPX-pages. I'm not sure why you are getting this error only after publishing. Perhaps your publishing process adds `VB` to the config or something similair?
2. Publishing: specific guidance is difficult because everyone does publishing differently. The default setup in the WingtipToys sample leaves your project just like it was with ASP.NET WebForms in .NET Framework so you can apply the same publishing logic *). If you wish to update your publishing, the recommended practice is to follow the patterns established for ASP.NET Core. 
3. Index-page: if you do not specify a start-page, ASP.NET Core per default requries an index page. In the Wingtip-Toys sample we don't change the ASP.NET Core default beut rather redirect from there. It's possible to set a different default via ASP.NET Core routing.

*) the one caveat being that the config-file is placed next to the executable in the bin folder but then you need to take the entire bin folder anyway.

Hope this helps!
~Michael
